src/App.tsx

```diff
@@ -1,16 +1,39 @@
 import React from 'react';
-
 import './App.scss';
 
-// import postsFromServer from './api/posts';
-// import commentsFromServer from './api/comments';
-// import usersFromServer from './api/users';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { PostList } from './components/PostList';
+
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const posts: Post[] = postsFromServer.map((post) => ({
+  ...post,
+  user: getUser(post.userId),
+  comments: [...commentsFromServer]
+    .filter((comment) => comment.postId === post.id)
+    .map((comment) => ({
+      ...comment,
+      user: getUser(post.userId),
```

Comment:
you don't need a user inside each comment (you don't use it)

Reply:
Done